### DISCUSSION TITLE
Malagu 3.0 规划讨论

### DISCUSSION THREAD

**Discussion by muxiangqiu** [cellbang/malagu#210] [Ideas]
Malagu 3.0 应该包含哪些重点特性？

**Comment by muxiangqiu** [comment]
我希望有以下事项：

1. AI 基础能力的支持
2. AI Agent 支持
3. 改名为 @celljs/xxx
4. 工具链与 AI 结合
5. 完善模块文档
7. 补充单元测试
8. 前端支持增强

**Comment by zhangfisher** [comment]
## 云原生开发套件

`malagu`作为一个云原生开发框架是优秀的，但是易用性还不够，功能集中在基础开发框架上，面向普通用户是不友好，特别是一些开箱即用的套件没有。

当打开阿里云、腾讯云时，面对一砣的概念，普通开发者往往是容易懵逼的，并不是很易用。

`malagu`可以开发一套更易用的`云原生开发套件`，具有简单的开发管理和部署界面，更贴近一般开发者，让云原生开发的门槛更加低。

可以参照对比一下`sealos`和`laf`和阿里云、腾讯云的云原生开发，我觉得这对普通开发者而言，更加友好。

再参照一个`vercal`和`nextjs`。

总之，我觉得和阿里云、腾讯云的云原生开发的门槛还是比较高的，这是`malagu`可以发力的地方

借用`sealos`的一句话：

阿里云、腾讯云云原生生态之庞大与复杂，无疑让许多企业感到手足无措，这个生态依然缺乏好用的、开箱即用的开发框架，就如同`vercal`和`nextjs`。

**Comment by zhangfisher** [comment]
## 基于文件的路由

相对于显式的`依赖注入`，约定大于配置的形式往往更加简洁，更香。
事实上，无论是前端或后端，引入基于文件的路由，会更加易用

- `controllers` : 后端的文件路由
- `pages` : 前端文件路由

**Reply by fsy0718** [reply]
前端静态资源引用也可以约定大于配置

**Comment by zhangfisher** [comment]
`sst.dev`值得学习
